src/datasets/filesystems/compression/gzip.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+import fsspec
+from fsspec.archive import AbstractArchiveFileSystem
+from fsspec.utils import DEFAULT_BLOCK_SIZE
+
+
+class GZipFileSystem(AbstractArchiveFileSystem):
+    """Read contents of GZIP archive as a file-system with one file inside."""
+
+    root_marker = ""
+    protocol = "gzip"
+
+    def __init__(
+        self,
+        fo="",
```

Comment:
question: in other places in `datasets` we use typing for each function argument. is there a reason we don't do that here?

Reply:
The code style comes from fsspec. I added the types